File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/GetImageListUseCaseImpl.java
package com.hwonchul.movie.domain.usecase.details;

import com.hwonchul.movie.domain.model.Image;
import com.hwonchul.movie.domain.repository.ImageRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.rxjava3.annotations.NonNull;
import io.reactivex.rxjava3.core.Flowable;

public class GetImageListUseCaseImpl implements GetImageListUseCase {
    private final ImageRepository repository;

    @Inject
    public GetImageListUseCaseImpl(@NonNull ImageRepository repository) {
        this.repository = repository;
    }

    @Override
    public Flowable<List<Image>> invoke(int movieId) {
        return repository.getAllPhotosByMovieId(movieId);
    }
}

File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/GetImageListUseCaseImpl.kt
package com.hwonchul.movie.domain.usecase.details

import com.hwonchul.movie.domain.model.Image
import com.hwonchul.movie.domain.repository.ImageRepository
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.catch
import kotlinx.coroutines.flow.map
import javax.inject.Inject

class GetImageListUseCaseImpl @Inject constructor(
    private val repository: ImageRepository
) : GetImageListUseCase {

    override suspend fun invoke(movieId: Int): Flow<Result<List<Image>>> =
        repository.getAllPhotosByMovieId(movieId)
            .map { Result.success(it) }
            .catch { emit(Result.failure(it)) }
}